internal/worker/remoterelations/relationunitsworker.go

```diff
@@ -27,59 +28,130 @@ type RelationUnitChangeEvent struct {
 	params.RemoteRelationChangeEvent
 }
 
+// Mode indicates whether the relation units worker is for a local or remote
+// model.
+type Mode string
+
+const (
+	// ModeLocal indicates that the worker is for a local model.
+	ModeLocal Mode = "local"
+	// ModeRemote indicates that the worker is for a remote model.
+	ModeRemote Mode = "remote"
+)
+
 // relationUnitsWorker uses instances of watcher.RelationUnitsWatcher to
 // listen to changes to relation settings in a model, local or remote.
 // Local changes are exported to the remote model.
 // Remote changes are consumed by the local model.
 type relationUnitsWorker struct {
 	catacomb catacomb.Catacomb
 
-	mu sync.Mutex
+	mode Mode
 
 	// mostRecentChange is stored here for the engine report.
+	mu               sync.Mutex
 	mostRecentChange RelationUnitChangeEvent
 	changeSince      time.Time
 
 	relationTag names.RelationTag
-	rrw         watcher.RemoteRelationWatcher
+	watcher     watcher.RemoteRelationWatcher
 	changes     chan<- RelationUnitChangeEvent
-	macaroon    *macaroon.Macaroon
-	mode        string // mode is local or remote.
 
+	mapEvent func(RelationUnitChangeEvent) RelationUnitChangeEvent
+
+	clock  clock.Clock
 	logger logger.Logger
 }
 
-func newRelationUnitsWorker(
+func newLocalRelationUnitsWorker(
+	ctx context.Context,
+	facade RemoteRelationsFacade,
 	relationTag names.RelationTag,
-	macaroon *macaroon.Macaroon,
-	rrw watcher.RemoteRelationWatcher,
+	mac *macaroon.Macaroon,
 	changes chan<- RelationUnitChangeEvent,
+	clock clock.Clock,
 	logger logger.Logger,
-	mode string,
-) (*relationUnitsWorker, error) {
+) (ReportableWorker, error) {
+	// Start a watcher to track changes to the units in the relation in the
+	// local model.
+	watcher, err := facade.WatchLocalRelationChanges(ctx, relationTag.Id())
```

Comment:
I assume we'll be swapping this out for a service-direct method.

Reply:
Yes.